app/js/components/LabResultEntry.jsx

```diff
@@ -80,9 +81,25 @@ export class LabResultEntry extends PureComponent {
   renderForm(selectedLabConcept) {
     const { patientHeaderDetail } = this.state;
     const {
-      CONSTANTS, conceptMembers, history: { location: { state } },
+      CONSTANTS, conceptMembers, patients, history: { location: { state } },
     } = this.props;
 
+    const selectedPatient = patients[patientHeaderDetail.uuid] || {};
+    const { encounters = [] } = selectedPatient;
```

Comment:
Does our current REST representation of the patient that we fetch include *all* the encounters for that patient?  I'm not sure that it does, but, if it does, I think it will be way too much to load... patients can have hundreds of encounters.

I think we will need to use the encounter endpoint where you can pass in an obs value that @dsurrao  created... in thi case you'd pass in the lab results test order concept and it's value:

https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/EncounterSearchHandler1_9.java#L42

So obsConcept would be the CONSTANTS.labResultsTestOrderNumberConcept and obsValue would be state.orderNumber

Reply:
Let me know if you think this ^^ will work @rhenshaw56 ....

Reply:
@mogoodrich, so currently we're hitting the encounter rest by encounterType of `labResultsEncounterType` and it just returns only encounters for test at the moment. I don't think other encounters are being returned, I also agree it's too much if other encounters were included in the response. do you think it would make sense to add this particular endpoint in as well, say add another method in the enconuterRest for react-component called `fetchEncounterByOBs`

Reply:
Ah, okay @rhenshaw56 .... limiting it to just lab results encounters should be fine for now (we will probably want to expand this in the future).

If if you only need a single encounter, the one that matches a specific order, then I'd use the REST endpoint @dsurrao created, and, yes, adding a fetchEncounterByObs method to encounterRest would be the way to do this.